browser/ntp_sponsored_images/view_counter_model.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ntp_sponsored_images/view_counter_model.h"
+
+#include "base/logging.h"
+
+namespace {
+
+constexpr int kInitialCountToBrandedWallpaper = 1;
+constexpr int kRegularCountToBrandedWallpaper = 3;
+
+}  // namespace
+
+ViewCounterModel::ViewCounterModel()
+    : count_to_branded_wallpaper_(kInitialCountToBrandedWallpaper) {
+}
+
+ViewCounterModel::~ViewCounterModel() = default;
+
+bool ViewCounterModel::ShouldShowBrandedWallpaper() const {
+  return count_to_branded_wallpaper_ == 0;
+}
+
+void ViewCounterModel::ResetCurrentWallpaperImageIndex() {
+  current_wallpaper_image_index_ = 0;
+}
+
+void ViewCounterModel::RegisterPageView() {
+  DCHECK_NE(-1, total_image_count_);
+
+  // When count is `0` then UI is free to show
+  // the branded wallpaper, until the next time `RegisterPageView`
+  // is called.
+  // We select the appropriate image index for the scheduled
+  // view of the branded wallpaper.
+  count_to_branded_wallpaper_--;
```

Comment:
seems like it would be simpler to just use a profile pref for this instead an extra class? Do we want this counter to always reset when the browser restarts?

Reply:
@bridiver yes, I believe so. Per the spec, branded new tab page should show up after opening / refreshing new tab page twice. After that, it's back to every 4th

Reply:
well it's easy enough to reset the counter when the service starts